core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -84,12 +85,15 @@ private DBWriter() {
             if (media != null) {
                 Log.i(TAG, String.format("Requested to delete FeedMedia [id=%d, title=%s, downloaded=%s",
                         media.getId(), media.getEpisodeTitle(), String.valueOf(media.isDownloaded())));
-                boolean result = false;
                 if (media.isDownloaded()) {
                     // delete downloaded media file
                     File mediaFile = new File(media.getFile_url());
                     if (mediaFile.exists()) {
-                        result = mediaFile.delete();
+                        if (!mediaFile.delete()) {
+                            Toast.makeText(context, "Unable to delete file. Rebooting the device could help.",
+                                    Toast.LENGTH_LONG).show();
```

Comment:
You could also send an `MessageEvent` which will be picked up here: https://github.com/AntennaPod/AntennaPod/blob/a89dd28e23eb7f2846aaed868b9c1f4755ccdf51/app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java#L738

Do trigger the event, I would just invert the if and revert it to its current behavior later.